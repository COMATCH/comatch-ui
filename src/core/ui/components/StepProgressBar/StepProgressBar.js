import React from 'react';
import classNames from 'classnames';

import {
    StyledLineWrapper,
    StyledProgressWrapper,
    StyledStepCircleWrapper,
    StyledStepTextWrapper,
    StyledStepWrapper,
    StyledWrapper,
} from './StyledWrapper';

const propTypes = {
    steps(props, propName) {
        if (!Array.isArray(props[propName])) {
            return new Error('StepProgressBar requires an array as steps prop');
        }
        if (props[propName].length < 2 || props[propName].length > 4) {
            return new Error('StepProgressBar must take between 2 and 4 steps');
        }

        let typeError;
        props[propName].forEach((step) => {
            if (typeof step !== 'string' && typeof step !== 'object') {
                typeError = 'every step passed to StepProgressBar must be a string or a <Translate /> instance';
            }
        });
        if (typeError) {
            return new Error(typeError);
        }
        return false;
    },
    currentStepIndex(props, propName) {
        if (!Number.isInteger(props[propName])) {
            return new Error('StepProgressBar requires a number as currentStepIndex prop');
        }
        if (props[propName] < 1 || props[propName] > props.steps.length) {
            return new Error('StepProgressBar currentStepIndex out of bounds');
        }
        return false;
    },
};

function renderSteps(steps, currentStepIndex) {
    return steps.map((step, index) => {
        const stepNumber = index + 1;
        const isActive = currentStepIndex > stepNumber || currentStepIndex === stepNumber;
        const classnames = classNames('StepProgressBar__step', `StepProgressBar__step${stepNumber}`, {
            active: stepNumber === currentStepIndex,
        });
        return (
            <StyledStepWrapper className={classnames} key={index}>
                <StyledStepCircleWrapper isActive={isActive} className="StepProgressBar__circle" />
                <StyledStepTextWrapper isActive={isActive} className="StepProgressBar__step-text">
                    {step}
                </StyledStepTextWrapper>
            </StyledStepWrapper>
        );
    });
}

export const StepProgressBar = ({ steps, currentStepIndex }) => {
    const classes = classNames(
        'StepProgressBar',
        `StepProgressBar__nb-steps-${steps.length}`,
        `StepProgressBar__step-active-${currentStepIndex}`,
    );

    return (
        <StyledWrapper numberOfSteps={steps.length} className={classes}>
            <StyledLineWrapper className="StepProgressBar__line">
                <StyledProgressWrapper
                    className="StepProgressBar__progress"
                    currentStepIndex={currentStepIndex}
                    numberOfSteps={steps.length}
                />
            </StyledLineWrapper>
            {renderSteps(steps, currentStepIndex)}
        </StyledWrapper>
    );
};

StepProgressBar.propTypes = propTypes;
