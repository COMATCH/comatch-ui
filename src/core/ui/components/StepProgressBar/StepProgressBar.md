## StepProgressBar

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**currentStepIndex** | `(custom validator)` |  | :x: | 
**steps** | `(custom validator)` |  | :x: | 

