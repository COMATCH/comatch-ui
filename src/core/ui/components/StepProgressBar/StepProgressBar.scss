@import '../../../styles/variables.scss';
@import '../../../styles/mixins.scss';

$progressColor: $co-color-primary;
$baseLineColor: $co-color-mid-gray;

$stepWidthRatio: 1/4;
$stepHalfWidthRatio: $stepWidthRatio / 2;

$stepWidth: 100% * $stepWidthRatio;
$stepHalfWidth: 100% * $stepHalfWidthRatio;

.StepProgressBar {

    height: 80px;
    position: relative;

    @include mobile {
        height: 95px;
    }

    &__line {
        width: 100%;
        height: 3px;
        background-color: $baseLineColor;
        position: absolute;
        top: 10px;
    }

    &__progress {
        position: absolute;
        width: 100%;
        top: 0;
        left: 0;
        bottom: 0;
        background-color: $progressColor;
        transition: transform 0.5s ease-out;
        transform-origin: left;
    }

    &__step {
        width: $stepWidth;
        position: absolute;
        top: 10px;

        &.active ~ .StepProgressBar__step {
            color: $co-color-mid-gray;
            font-weight: bold;
        }
    }

    &__step-text {
        text-align: center;
        padding-top: 15px;
    }

    &__circle {
        width: 10px;
        height: 10px;
        position: absolute;
        background: $baseLineColor;
        left: calc(50% - 5px);
        border-radius: 100%;
        top: -3px;
    }

    &__nb-steps-2 {
        .StepProgressBar__step1 {
            left: 0;
        }

        .StepProgressBar__step2 {
            left: calc(100% - #{$stepWidth});
        }

        &.StepProgressBar__step-active-1 {
            .StepProgressBar__progress {
                transform: scaleX($stepHalfWidthRatio);
            }
        }

        &.StepProgressBar__step-active-2 {
            .StepProgressBar__progress {
                transform: scaleX(1 - $stepHalfWidthRatio);
            }
        }

        @include mobile {
            $largeStepWidthRatio: 0.35;
            $largeStepWidth: 100% * $largeStepWidthRatio;

            .StepProgressBar__step {
                width: $largeStepWidth;
            }
            .StepProgressBar__step-text {
                padding-top: 10px;
            }
            .StepProgressBar__step2 {
                left: calc(100% - #{$largeStepWidth});
            }

            &.StepProgressBar__step-active-1 {
                .StepProgressBar__progress {
                    transform: scaleX($largeStepWidthRatio / 2);
                }
            }

            &.StepProgressBar__step-active-2 {
                .StepProgressBar__progress {
                    transform: scaleX(1 - ($largeStepWidthRatio / 2));
                }
            }
        }
    }
    &__nb-steps-3 {
        .StepProgressBar__step1 {
            left: 0;
        }

        .StepProgressBar__step2 {
            left: calc(50% - #{$stepHalfWidth});
        }

        .StepProgressBar__step3 {
            left: calc(100% - #{$stepWidth});
        }

        &.StepProgressBar__step-active-1 {
            .StepProgressBar__progress {
                transform: scaleX($stepHalfWidthRatio);
            }
        }

        &.StepProgressBar__step-active-2 {
            .StepProgressBar__progress {
                transform: scaleX(0.5);
            }
        }

        &.StepProgressBar__step-active-3 {
            .StepProgressBar__progress {
                transform: scaleX(1 - $stepHalfWidthRatio);
            }
        }
    }
    &__nb-steps-4 {
        .StepProgressBar__step1 {
            left: 0;
        }

        .StepProgressBar__step2 {
            left: calc((100% - #{$stepWidth}) / 3);
        }

        .StepProgressBar__step3 {
            left: calc((100% - #{$stepWidth}) / 3 * 2);
        }

        .StepProgressBar__step4 {
            left: calc(100% - #{$stepWidth});
        }

        &.StepProgressBar__step-active-1 {
            .StepProgressBar__progress {
                transform: scaleX($stepHalfWidthRatio);
            }
        }

        &.StepProgressBar__step-active-2 {
            .StepProgressBar__progress {
                transform: scaleX(((1 - $stepWidthRatio) / 3) + $stepHalfWidthRatio);
            }
        }

        &.StepProgressBar__step-active-3 {
            .StepProgressBar__progress {
                transform: scaleX(((1 - $stepWidthRatio) / 3 * 2) + $stepHalfWidthRatio);
            }
        }

        &.StepProgressBar__step-active-4 {
            .StepProgressBar__progress {
                transform: scaleX((1 - $stepWidthRatio) + $stepHalfWidthRatio);
            }
        }

        .StepProgressBar__step-text {
            padding-left: 4%;
            padding-right: 4%;
        }
    }

    &__step-active-1 {
        .StepProgressBar__step1 .StepProgressBar__circle {
            background-color: $progressColor;
        }

        .StepProgressBar__step1 .StepProgressBar__step-text {
            font-weight: bold;
        }
    }
    &__step-active-2 {
        .StepProgressBar__step1 .StepProgressBar__circle,
        .StepProgressBar__step2 .StepProgressBar__circle {
            background-color: $progressColor;
        }

        .StepProgressBar__step2 .StepProgressBar__step-text {
            font-weight: bold;
        }
    }
    &__step-active-3 {
        .StepProgressBar__step1 .StepProgressBar__circle,
        .StepProgressBar__step2 .StepProgressBar__circle,
        .StepProgressBar__step3 .StepProgressBar__circle {
            background-color: $progressColor;
        }

        .StepProgressBar__step3 .StepProgressBar__step-text {
            font-weight: bold;
        }
    }
    &__step-active-4 {
        .StepProgressBar__step1 .StepProgressBar__circle,
        .StepProgressBar__step2 .StepProgressBar__circle,
        .StepProgressBar__step3 .StepProgressBar__circle,
        .StepProgressBar__step4 .StepProgressBar__circle {
            background-color: $progressColor;
        }

        .StepProgressBar__step4 .StepProgressBar__step-text {
            font-weight: bold;
        }
    }
}
