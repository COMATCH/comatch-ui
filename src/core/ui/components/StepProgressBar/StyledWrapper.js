import React from 'react';
import styled from 'styled-components';
import { device } from '../../../styles/mixins';
import { palette } from '../../../styles/variables';

const BASE_LINE_COLOR = palette.midGray;
const PROGRESS_COLOR = palette.primary;

const STEP_WIDTH_RATIO = 1 / 4;
const STEP_HALF_WIDTH_RATIO = STEP_WIDTH_RATIO / 2;

const STEP_WIDTH = `calc(100% * ${STEP_WIDTH_RATIO})`;
const STEP_HALF_WIDTH = `100% * ${STEP_HALF_WIDTH_RATIO}`;

const calculateProgressLineScale = ({ currentStepIndex, numberOfSteps }) => {
    if (!numberOfSteps || numberOfSteps === 1 || !currentStepIndex) {
        return '';
    }

    if (numberOfSteps === 2) {
        return currentStepIndex === 1 ? 'transform: scaleX(0);' : '';
    }

    if (numberOfSteps === 3) {
        switch (currentStepIndex) {
            case 1:
                return 'transform: scaleX(0);';

            case 2:
                return 'transform: scaleX(0.5);';

            default:
                return '';
        }
    } else if (numberOfSteps === 4) {
        switch (currentStepIndex) {
            case 1:
                return 'transform: scaleX(0);';

            case 2:
                return 'transform: scaleX(0.333);';

            case 3:
                return 'transform: scaleX(0.666);';

            default:
                return '';
        }
    } else {
        return '';
    }
};

const generateNumberOfStepsStyling = ({ numberOfSteps }) => {
    switch (numberOfSteps) {
        case 2:
            return `
                .StepProgressBar__step1 {
                    left: 0;
                }

                .StepProgressBar__step2 {
                    left: calc(100% - ${STEP_WIDTH});
                }
            `;

        case 3:
            return `
                .StepProgressBar__step1 {
                    left: 0;
                }

                .StepProgressBar__step2 {
                    left: calc(50% - calc(${STEP_HALF_WIDTH}));
                }

                .StepProgressBar__step3 {
                    left: calc(100% - ${STEP_WIDTH});
                }
            `;

        case 4:
            return `
                .StepProgressBar__step1 {
                    left: 0;
                }

                .StepProgressBar__step2 {
                    left: calc((100% - ${STEP_WIDTH}) / 3);
                }

                .StepProgressBar__step3 {
                    left: calc((100% - ${STEP_WIDTH}) / 3 * 2);
                }

                .StepProgressBar__step4 {
                    left: calc(100% - ${STEP_WIDTH});
                }

                .StepProgressBar__step-text {
                    padding-left: 4%;
                    padding-right: 4%;
                }
            `;

        default:
            return '';
    }
};

const generateStepCircleActiveStyling = ({ isActive }) => (!isActive ? '' : `background-color: ${PROGRESS_COLOR};`);

const generateStepTextActiveStyling = ({ isActive }) => (!isActive ? '' : 'font-weight: bold;');

export const StyledStepCircleWrapper = styled(
    React.forwardRef(({ isActive, ...rest }, ref) => <div {...rest} ref={ref} />),
)`
    width: 10px;
    height: 10px;
    position: absolute;
    background: ${BASE_LINE_COLOR};
    left: calc(50% - 5px);
    border-radius: 100%;
    top: -3px;

    ${generateStepCircleActiveStyling}
`;

export const StyledStepTextWrapper = styled(
    React.forwardRef(({ isActive, ...rest }, ref) => <div {...rest} ref={ref} />),
)`
    text-align: center;
    padding-top: 15px;

    ${generateStepTextActiveStyling}
`;

export const StyledStepWrapper = styled.div`
    width: ${STEP_WIDTH};
    position: absolute;
    top: 10px;

    &.active ~ .StepProgressBar__step {
        color: ${BASE_LINE_COLOR};
        font-weight: bold;
    }
`;

export const StyledProgressWrapper = styled(
    React.forwardRef(({ currentStepIndex, numberOfSteps, ...rest }, ref) => <div {...rest} ref={ref} />),
)`
    position: absolute;
    width: 100%;
    top: 0;
    left: 0;
    bottom: 0;
    background-color: ${PROGRESS_COLOR};
    transition: transform 0.5s ease-out;
    transform-origin: left;

    ${calculateProgressLineScale}
`;

export const StyledLineWrapper = styled.div`
    width: 100%;
    height: 3px;
    background-color: ${BASE_LINE_COLOR};
    position: absolute;
    top: 10px;

    width: 75%;
    transform: translateX(17%);
`;

export const StyledWrapper = styled(React.forwardRef(({ numberOfSteps, ...rest }, ref) => <div {...rest} ref={ref} />))`
    height: 95px;
    position: relative;

    @media ${device.desktop} {
        height: 80px;
    }

    ${generateNumberOfStepsStyling}
`;
