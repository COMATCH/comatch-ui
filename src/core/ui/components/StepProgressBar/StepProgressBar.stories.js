import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { StepProgressBar } from './StepProgressBar';

storiesOf('StepProgressBar', module)
    .add('with 2 steps, first one active', () => (
        <>
            <StepProgressBar
                steps={['this is the first step', 'this is the second and last step']}
                currentStepIndex={1}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    `\n\tsteps={['this is the first step', 'this is the second and last step']}` +
                    '\n\tcurrentStepIndex={1}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 2 steps, second one active', () => (
        <>
            <StepProgressBar
                steps={['this is the first step', 'this is the second and last step']}
                currentStepIndex={2}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    `\n\tsteps={['this is the first step', 'this is the second and last step']}` +
                    '\n\tcurrentStepIndex={2}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 3 steps, first one active', () => (
        <>
            <StepProgressBar
                steps={['this is the first step', 'this is the second step', 'this is the third and last step']}
                currentStepIndex={1}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    `\n\tsteps={['this is the first step', 'this is the second step',` +
                    ` 'this is the third and last step']}` +
                    '\n\tcurrentStepIndex={1}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 3 steps, second one active', () => (
        <>
            <StepProgressBar
                steps={['this is the first step', 'this is the second step', 'this is the third and last step']}
                currentStepIndex={2}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    `\n\tsteps={['this is the first step', 'this is the second step',` +
                    ` 'this is the third and last step']}` +
                    '\n\tcurrentStepIndex={2}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 3 steps, third one active', () => (
        <>
            <StepProgressBar
                steps={['this is the first step', 'this is the second step', 'this is the third and last step']}
                currentStepIndex={3}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    `\n\tsteps={['this is the first step', 'this is the second step',` +
                    ` 'this is the third and last step']}` +
                    '\n\tcurrentStepIndex={3}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 4 steps, first one active', () => (
        <>
            <StepProgressBar
                steps={[
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ]}
                currentStepIndex={1}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    '\n\tsteps={[' +
                    `\n\t\t'this is the first step',` +
                    `\n\t\t'this is the second step',` +
                    `\n\t\t'this is the third step',` +
                    `\n\t\t'this is the fourth and last step',` +
                    '\n\t]}' +
                    '\n\tcurrentStepIndex={1}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 4 steps, second one active', () => (
        <>
            <StepProgressBar
                steps={[
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ]}
                currentStepIndex={2}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    '\n\tsteps={[' +
                    `\n\t\t'this is the first step',` +
                    `\n\t\t'this is the second step',` +
                    `\n\t\t'this is the third step',` +
                    `\n\t\t'this is the fourth and last step',` +
                    '\n\t]}' +
                    '\n\tcurrentStepIndex={2}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 4 steps, third one active', () => (
        <>
            <StepProgressBar
                steps={[
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ]}
                currentStepIndex={3}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    '\n\tsteps={[' +
                    `\n\t\t'this is the first step',` +
                    `\n\t\t'this is the second step',` +
                    `\n\t\t'this is the third step',` +
                    `\n\t\t'this is the fourth and last step',` +
                    '\n\t]}' +
                    '\n\tcurrentStepIndex={3}' +
                    '\n/>'}
            </Highlight>
        </>
    ))
    .add('with 4 steps, last one active', () => (
        <>
            <StepProgressBar
                steps={[
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ]}
                currentStepIndex={4}
            />
            <Highlight className="html">
                {'<StepProgressBar' +
                    '\n\tsteps={[' +
                    `\n\t\t'this is the first step',` +
                    `\n\t\t'this is the second step',` +
                    `\n\t\t'this is the third step',` +
                    `\n\t\t'this is the fourth and last step',` +
                    '\n\t]}' +
                    '\n\tcurrentStepIndex={4}' +
                    '\n/>'}
            </Highlight>
        </>
    ));
