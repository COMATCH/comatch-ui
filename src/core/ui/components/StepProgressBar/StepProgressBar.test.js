import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { StepProgressBar } from './StepProgressBar';
import '../../../config/tests/setup';

describe('StepProgressBar tests', () => {
    const REQUIRED_PROPS = {
        steps: [
            'this is the first step',
            'this is the second step',
            'this is the third step',
            'this is the fourth and last step',
        ],
        currentStepIndex: 4,
    };

    it('should render and matched values in snapshot', () => {
        const stepProgressBarComponent = renderer.create(<StepProgressBar {...REQUIRED_PROPS} />).toJSON();
        expect(stepProgressBarComponent).toMatchSnapshot();
    });

    it('should render correctly with required props', () => {
        const props = {
            ...REQUIRED_PROPS,
        };
        const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
        expect(stepProgressBarComponent.prop('steps')).toEqual(props.steps);
        expect(stepProgressBarComponent.prop('currentStepIndex')).toEqual(props.currentStepIndex);
    });

    describe('verifying the HTML structure', () => {

        it('should render correctly with 2 steps, first one active', () => {
            const props = {
                steps: ['this is the first step', 'this is the second step'],
                currentStepIndex: 1,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(
                stepProgressBarComponent.find('.StepProgressBar__nb-steps-2').at(0)
                    .hasClass('StepProgressBar__step-active-1')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(6)
                .text()).toEqual('this is the second step');
        });

        it('should render correctly with 2 steps, second one active', () => {
            const props = {
                steps: ['this is the first step', 'this is the second step'],
                currentStepIndex: 2,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-2').at(0)
                .hasClass('StepProgressBar__step-active-2')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(6)
                .text()).toEqual('this is the second step');
        });

        it('should render correctly with 3 steps, first one active', () => {
            const props = {
                steps: ['this is the first step', 'this is the second step', 'this is the third step'],
                currentStepIndex: 1,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-3').at(0)
                .hasClass('StepProgressBar__step-active-1')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(9)
                .text()).toEqual('this is the third step');
        });

        it('should render correctly with 3 steps, second one active', () => {
            const props = {
                steps: ['this is the first step', 'this is the second step', 'this is the third step'],
                currentStepIndex: 2,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-3').at(0)
                .hasClass('StepProgressBar__step-active-2')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(9)
                .text()).toEqual('this is the third step');
        });

        it('should render correctly with 3 steps, third one active', () => {
            const props = {
                steps: ['this is the first step', 'this is the second step', 'this is the third step'],
                currentStepIndex: 3,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-3').at(0)
                .hasClass('StepProgressBar__step-active-3')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(9)
                .text()).toEqual('this is the third step');
        });

        it('should render correctly with 4 steps, first one active', () => {
            const props = {
                steps: [
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ],
                currentStepIndex: 1,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-4').at(0)
                .hasClass('StepProgressBar__step-active-1')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(12)
                .text()).toEqual('this is the fourth and last step');
        });

        it('should render correctly with 4 steps, second one active', () => {
            const props = {
                steps: [
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ],
                currentStepIndex: 2,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-4').at(0)
                .hasClass('StepProgressBar__step-active-2')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(12)
                .text()).toEqual('this is the fourth and last step');
        });

        it('should render correctly with 4 steps, third one active', () => {
            const props = {
                steps: [
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ],
                currentStepIndex: 3,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-4').at(0)
                .hasClass('StepProgressBar__step-active-3')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(12)
                .text()).toEqual('this is the fourth and last step');
        });

        it('should render correctly with 4 steps, last one active', () => {
            const props = {
                steps: [
                    'this is the first step',
                    'this is the second step',
                    'this is the third step',
                    'this is the fourth and last step',
                ],
                currentStepIndex: 4,
            };
            const stepProgressBarComponent = mount(<StepProgressBar {...props} />);
            expect(stepProgressBarComponent.find('.StepProgressBar__nb-steps-4').at(0)
                .hasClass('StepProgressBar__step-active-4')).toBe(true);
            expect(stepProgressBarComponent.find('.StepProgressBar__step-text').at(12)
                .text()).toEqual('this is the fourth and last step');
        });
    });
});
