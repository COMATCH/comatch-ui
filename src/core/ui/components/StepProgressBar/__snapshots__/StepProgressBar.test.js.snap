// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepProgressBar tests should render and matched values in snapshot 1`] = `
<div
  className="sc-EHOje bsmZDm StepProgressBar StepProgressBar__nb-steps-4 StepProgressBar__step-active-4"
>
  <div
    className="sc-ifAKCX GHHLr StepProgressBar__line"
  >
    <div
      className="sc-bxivhb cSxQvU StepProgressBar__progress"
    />
  </div>
  <div
    className="sc-htpNat SegEG StepProgressBar__step StepProgressBar__step1"
  >
    <div
      className="sc-bdVaJa haIpKu StepProgressBar__circle"
    />
    <div
      className="sc-bwzfXH lfzIFG StepProgressBar__step-text"
    >
      this is the first step
    </div>
  </div>
  <div
    className="sc-htpNat SegEG StepProgressBar__step StepProgressBar__step2"
  >
    <div
      className="sc-bdVaJa haIpKu StepProgressBar__circle"
    />
    <div
      className="sc-bwzfXH lfzIFG StepProgressBar__step-text"
    >
      this is the second step
    </div>
  </div>
  <div
    className="sc-htpNat SegEG StepProgressBar__step StepProgressBar__step3"
  >
    <div
      className="sc-bdVaJa haIpKu StepProgressBar__circle"
    />
    <div
      className="sc-bwzfXH lfzIFG StepProgressBar__step-text"
    >
      this is the third step
    </div>
  </div>
  <div
    className="sc-htpNat SegEG StepProgressBar__step StepProgressBar__step4 active"
  >
    <div
      className="sc-bdVaJa haIpKu StepProgressBar__circle"
    />
    <div
      className="sc-bwzfXH lfzIFG StepProgressBar__step-text"
    >
      this is the fourth and last step
    </div>
  </div>
</div>
`;
